merkle/crdt/composite.go

```diff
@@ -64,25 +32,20 @@ type MerkleCompositeDAG struct {
 // NewMerkleCompositeDAG creates a new instance (or loaded from DB) of a MerkleCRDT
 // backed by a CompositeDAG CRDT.
 func NewMerkleCompositeDAG(
-	datastore datastore.DSReaderWriter,
-	headstore datastore.DSReaderWriter,
-	dagstore datastore.DAGStore,
+	txn datastore.Txn,
 	schemaVersionKey core.CollectionSchemaVersionKey,
-	uCh events.UpdateChannel,
-	ns,
 	key core.DataStoreKey,
 	fieldName string,
```

Comment:
suggestion: It feels like (I haven't chased down the full code flow) that `fieldName` could also be removed as a parameter here?  And possibly removed from the core/crdt.`CompositeDAG` struct. On a composite commit this will always be `"C"` no?

Reply:
Yeah I though about doing that but I was trying to keep the arguments the same for both merkle crdt types to see if we could eventually simplify things even further. I'll give it some more thinking.

Reply:
Ah fair enough, don't go out of your way for this then, especially if it disrupts flow and/or you have other stuff based off of this branch.